src/neuroglancer/datasource/computed/frontend.ts

```diff
@@ -0,0 +1,258 @@
+  // URL of the origin volume chunk source, used as computational input.
+  originUrl: string;
+  // Index of the input scale on the origin source.
+  inputScaleIndex: number;
+  // Index of the input volume at the given scale on the origin source.
+  inputSourceIndex: number;
+  // Output voxel size.
+  outputVoxelSize: vec3;
+  // Output voxel offset.
+  outputVoxelOffset: vec3;
+  // Output volume size.
+  outputVolumeSize: vec3;
+  // Determines input buffer size as well as output data chunks size, etc.
+  computationParameters: ComputationParameters;
+
+  dataSourceProvider: DataSourceProvider;
+}
+
+interface ComputedVolumeSpecs {
+  sources: VolumeChunkSource[][];
+  volumeType: VolumeType;
+  dataType: DataType;
+  numChannels: number;
+}
+
+export class ComputedMultiscaleVolumeChunkSource implements GenericMultiscaleVolumeChunkSource {
+  numChannels: number;
+  dataType: DataType;
+  volumeType: VolumeType;
+
+  constructor(
+      public params: ComputedVolumeDataSourceParameters, public sources: VolumeChunkSource[][],
+      public computation: VolumeComputationFrontend<any>, public chunkManager: ChunkManager) {
+    this.numChannels = params.computationParameters.outputSpec.numChannels;
+    this.dataType = params.computationParameters.outputSpec.dataType;
+    this.volumeType = params.computationParameters.outputSpec.volumeType;
+  }
+
+  getSources(volumeSourceOptions: VolumeSourceOptions) {
+    const spec = VolumeChunkSpecification.getDefaults({
+      voxelSize: this.params.outputVoxelSize,
+      dataType: this.dataType,
+      numChannels: this.numChannels,
+      chunkDataSizes: [this.params.computationParameters.outputSpec.size],
+      transform: mat4.fromTranslation(
+          mat4.create(),
+          vec3.multiply(vec3.create(), this.params.outputVoxelSize, this.params.outputVoxelOffset)),
+      upperVoxelBound: this.params.outputVolumeSize,
+      volumeType: this.volumeType,
+      baseVoxelOffset: this.params.outputVoxelOffset,
+      volumeSourceOptions,
+    })[0];
+
+    const originSource = this.sources[this.params.inputScaleIndex][this.params.inputSourceIndex];
+    const inputResolution = originSource.spec.voxelSize;
+
+    const parameters: ComputedVolumeChunkSourceParameters = {
+      computationRef: this.computation.addCounterpartRef(),
+      sourceRef: originSource.addCounterpartRef(),
+      inputSize: this.params.computationParameters.inputSpec.size,
+      scaleFactor: vec3.divide(vec3.create(), this.params.outputVoxelSize, inputResolution)
+    };
+
+    const computedSource =
+        this.chunkManager.getChunkSource(ComputedVolumeChunkSource, {spec, parameters});
+
+    return [[computedSource]];
+  }
+
+  getMeshSource() {
+    return null;
+  }
+}
+
+
+export class ComputedDataSource extends DataSource {
+  parseConfig(config: string, {} /*dataSourceProvider: DataSourceProvider*/) {
+    // config is expected to be like computeName(computeParams)source(url)
```

Comment:
Clarify accepted syntax, perhaps with an example.

It also seems likely that the parameters will be sufficiently complicated that no one would type them in directly, in which case you could simplify things by just using JSON encoding of the whole thing.  The code in datasource/index.ts doesn't place any restrictions on the contents of the datasource "url" after the protocol:// part.

Reply:
That makes sense to me, and yes, computational datasources are likely to have fairly complicated configurations. What would you think about modifications to the UI to make it easier to manage those? (Not in this PR, though)

Reply:
Yeah, agreed that UI modifications would make sense in a future PR.